README.md

```diff
@@ -208,6 +209,37 @@ ros2 run topic_tools mux sel_cmdvel auto_cmdvel joystick_cmdvel
 - `initial_topic` (str, default="")
     - Input topic to select on startup. If `__none`, start with no input topic. If unset, default to first topic in arguments
 
+### Demux
+
+demux is a ROS2 node that subscribes to an incoming topic and republishes incoming data to one of many topic,
+i.e., it's a demultiplexer that switches an input towards 1 of N outputs. Services are offered to switch among output topics,
+and to add and delete output topics. At startup, the first output topic on the command line is selected.
+
+#### Usage
+
+```shell
+ros2 run topic_tools demux <intopic> <outtopic1> [outopic2...]
+```
+
+Subscribe to <intopic1> and publish currently to selected outopic among <outopic1>...N. demux will start with <outopic1> selected.
+- `intopic`: Incoming topic to subscribe to
+- `outopicN`: Outgoing topic to publish on
+
+E.g. demux one command stream (cmdvel) between two command streams (turtle1_cmdvel and turtle2_cmdvel):
+
+```shell
+ros2 run topic_tools demux cmdvel turtle1_cmdvel turtle2_cmdvel
+```
+
+#### Parameters
+
+- `input_topic` (string, default=`~/input`)
+    - the same as if provided as a command line argument
+- `output_topics` (string array)
+    - the same as if provided as a command line argument
+- `initial_topic` (str, default="")
+    - Output topic to select on startup. If `__none`, start with no output topic. If unset, default to first topic in arguments
+
```

Comment:
Having all 3 arguments as positional when one of them is a space-separated list is not a good design decision. It will be more cleaner to make all of them as explicit no-positional arguments with short alternatives. e.g. `--input-topic -i`, `--output-topics -o`.

BTW. What is the `initial_topic` parameter? I don't see that it was parsed or taking in to account?

Reply:
`initial_topic` is read here: https://github.com/rcywongaa/topic_tools/blob/add_demux/topic_tools/src/demux_node.cpp#L33

Are we open to adding dependency to a proper argparsing library?